Packs/Rapid7InsightVMCloud/Integrations/Rapid7InsightVMCloud/Rapid7InsightVMCloud.py

```diff
@@ -0,0 +1,455 @@
+
+    response = client.make_request(
+        method=method,
+        url_suffix=endpoint,
+        json_data=data
+    )
+    if response:
+        scan_response = response.get("scans")
+        markdown = tableToMarkdown(
+            'This is required Scan Result',
+            scan_response
+        )
+
+        result = CommandResults(
+            readable_output=markdown,
+            outputs_prefix='Rapid7.InsighVMCloud.Scans',
+            outputs_key_field='id',
+            outputs=scan_response
+        )
+        return result
+    else:
+        return_error(f'no response')
+
+
+def last_sites_command(client: Client, page: int, size=int) -> CommandResults:
+    """
+    Returns the details for sites.
+
+    Args:
+        client (Client Object): Client object with request.
+        page: The index of the page (zero-based) to retrieve
+        size: The number of records per page to retrieve.
+
+    """
+
+    args = demisto.args()
+    params = {
+        "page": args.get("page"),
+        "size": args.get("size")
+    }
+
+    if int(args.get("size")) > 500:
+        return_error("Exceed size limit")
+
+    else:
+        method = "POST"
+        endpoint = "/v4/integration/sites"
+        headers = ["name", "type"]
+        response = client.make_request(
+            method=method,
+            url_suffix=endpoint,
+            params=params
+        )
+        if response:
+            res = response.get("data")
+            markdown = tableToMarkdown('List Sites', res, headers)
+            result = CommandResults(
+                readable_output=markdown,
+                outputs_prefix='Rapid7.InsighVMCloud.Sites',
+                outputs_key_field='name',
+                outputs=response
+            )
+            return result
+        else:
+            return_error("no response")
+
+
+def search_vulnerabilities_command(client: Client, query: str, page: int, size=int) -> dict:
+    """
+    Returns all vulnerabilities that can be assessed.
+
+    Args:
+        client (Client Object): Client object with request.
+        query: query to search vulnerabilities.
+        page: The index of the page (zero-based) to retrieve
+        size: The number of records per page to retrieve.
+
+    """
+
+    args = demisto.args()
+    query = args.get("query")
+    params = {
+        "page": args.get("page"),
+        "size": args.get("size")
+    }
+    if int(args.get("size")) > 500:
+        return_error("Exceed size limit")
+    else:
+        method = "POST"
+        endpoint = "/v4/integration/vulnerabilities"
+        data = {
+            "vulnerability": query
+        }
+        response = client.make_request(
+            method=method,
+            url_suffix=endpoint,
+            json_data=data,
+            params=params
+        )
+
+        if response:
+            return response
+        else:
+            return_error('no response')
+
+
+def stop_scan_command(client: Client, id: str) -> CommandResults:
```

Comment:
Why do we need the variables `id: str` they are not used ? @thimanshu474

Reply:
It is used in line 395(scan_id = args.get("id")) and then it takes as an input in endpoint.

Reply:
You're using scan_id and you do args.get("id") in which id is a string but you don't use the variable id, am I missing something ?

Reply:
Now I have understand the issue, in that case we can delete these lines.                                                             
-args = demisto.args()
-scan_id = args.get("id")
and get value of id directly from command argument like this.
endpoint = f"/v4/integration/scan/{id}/stop"
then this issue is resolved.

Reply:
It won't solver it the variables in the function are still redundant page for example will be `args.get("page")` if it exist or `None` if it's not and the page in the function signature is meaningless `search_vulnerabilities_command(client: Client, query: str, page: int, size=int)` why do you need the variables in the signature if you're getting them from args ?
@thimanshu474

Reply:
Please delete line number 394,395 not taking variable from args.get()

def stop_scan_command(client:Client, id: str)-> CommandResults:

    method = "POST"
    endpoint = f"/v4/integration/scan/{id}/stop"